traceabilityViewer/app/views.py

```diff
@@ -317,4 +323,9 @@ def get_data_with_cypher_query(cypher_query):
                 raise TypeError(
                     f"Expected Node or Relationship type to be returned from the query; " f"got {type(element)}"
                 )
+    node_names = [node.name for node in node_objects]
+    for node in node_objects:
+        for link in node.links:
+            if link.source in node_names and link.target in node_names:
+                links.add(link)
```

Comment:
So, isn't there a more efficient solution than this?

Reply:
I don't know how yet. You don't know which nodes will be added next... Or I leave it as previous, the nodes and links are actually correct. These are the one you asked for via the Cypher query, but because the eye-icon looks to all the links of that node, the nodes that are connected are visible, even if it's a (hidden) link that is not returned by Cypher query.